tests/lsp_features/definition.zig

```diff
@@ -12,24 +11,139 @@ const offsets = zls.offsets;
 
 const allocator: std.mem.Allocator = std.testing.allocator;
 
-test "definition - smoke" {
+test "goto - global variable" {
     try testDefinition(
-        \\fn main() void { f<>oo(); }
-        \\fn <def>foo</def>() void {}
+        \\const <def><decl>foo</decl></def> = 5;
+        \\comptime {
+        \\    _ = <>foo;
+        \\}
+    );
+    try testDefinition(
+        \\const <def><decl>foo</decl></def>: <tdef>u32</tdef> = 5;
+        \\comptime {
+        \\    _ = <>foo;
+        \\}
+    );
+    try testDefinition(
+        \\const <def><decl><>foo</decl></def> = 5;
+    );
+    try testDefinition(
+        \\const <def><decl><>foo</decl></def>: <tdef>u32</tdef> = 5;
+    );
+}
+
+test "goto - local variable" {
+    try testDefinition(
+        \\comptime {
+        \\    var <def><decl>foo</decl></def> = 5;
+        \\    {
+        \\        var bar = 5;
+        \\        _ = <>foo;
+        \\        _ = bar;
+        \\    }
+        \\}
+    );
+    try testDefinition(
+        \\comptime {
+        \\    var foo = 5;
+        \\    {
+        \\        var <def><decl>bar</decl></def> = 5;
+        \\        _ = foo;
+        \\        _ = <>bar;
+        \\    }
+        \\}
+    );
+}
+
+test "goto - assign destructure" {
+    try testDefinition(
+        \\comptime {
+        \\    const foo, const <def><decl>bar</decl></def>: <tdef>u32</tdef> = .{ 1, 2 };
+        \\    _ = foo;
+        \\    _ = <>bar;
+        \\}
+    );
+}
+
+test "goto - function parameter" {
+    try testDefinition(
+        \\fn f(<def><decl>foo</decl></def>: <tdef>u32</tdef>) void {
+        \\    _ = <>foo;
+        \\}
+    );
+}
+
+test "goto - field access" {
+    try testDefinition(
+        \\const S = struct { <def><decl>alpha</decl></def>: <tdef>u32</tdef> };
+        \\var s: S = undefined;
+        \\const foo = s.<>alpha;
+    );
+}
+
+test "goto - struct init" {
+    try testDefinition(
+        \\const S = struct { <def><decl>alpha</decl></def>: <tdef>u32</tdef> };
+        \\var s = S{ .<>alpha = 5};
+    );
+}
+
+test "goto - capture" {
+    try testDefinition(
+        \\test {
+        \\    const S = <tdef>struct</tdef> {};
+        \\    var maybe: ?S = 5;
+        \\    if (maybe) |<>some| {}
+        \\}
+    );
+    if (true) return error.SkipZigTest; // TODO
+    // primitives like `u32` are represented as a InternPool.Index so they
+    // don't have a Ast.Node.Index that gives them a source location
+    try testDefinition(
+        \\test {
+        \\    var maybe: <tdef>?u32<&tdef> = 5;
```

Comment:
Is the `<&tdef>` a typo or special syntax?

Reply:
just a typo. The `&` and `/` are next to each other on a german keyboard.